genesis/deterministic_genesis.go

```diff
@@ -0,0 +1,98 @@
+package genesis
+
+import (
+	"math/rand"
+	"time"
+
+	"fmt"
+
+	acm "github.com/hyperledger/burrow/account"
+	"github.com/hyperledger/burrow/permission"
+	"github.com/tendermint/ed25519"
+	"github.com/tendermint/go-crypto"
+)
+
+type deterministicGenesis struct {
+	random *rand.Rand
+}
+
+// Generates deterministic pseudo-random genesis state
+func NewDeterministicGenesis(seed int64) *deterministicGenesis {
+	return &deterministicGenesis{
+		random: rand.New(rand.NewSource(seed)),
+	}
+}
+
+func (dg *deterministicGenesis) GenesisDoc(numAccounts int, randBalance bool, minBalance uint64, numValidators int,
+	randBonded bool, minBonded int64) (*GenesisDoc, []acm.PrivateAccount) {
+
+	accounts := make([]Account, numAccounts)
+	privAccounts := make([]acm.PrivateAccount, numAccounts)
+	defaultPerms := permission.DefaultAccountPermissions
+	for i := 0; i < numAccounts; i++ {
+		account, privAccount := dg.Account(randBalance, minBalance)
```

Comment:
why `randBalance` ...? I'm not sure I follow the logic of that.

Reply:
it's legacy from where this code was derived. It used to be truly random, I made is deterministically random. Really it doesn't need to be either, you could feasibly use it to do some automated testing over a range of seeds. I wouldn't have written it this way, and I possibly should have cleared it up when I rewrote parts but I wasn't clear on what tests were depending on so I gave option of picking some random values. I don't think it does much harm leaving it in for now.

Reply:
Cool.